cache/cache.go

```diff
@@ -6,29 +6,29 @@ import (
 	ibytes "github.com/cosmos/iavl/internal/bytes"
 )
 
-// Node[T] represents a node eligible for caching.
-type Node[T any] interface {
```

Comment:
what's the reason for this revert?

Reply:
Previously, the `global nonce` is used as a nodekey, so we didn't need to convert. But now we are using `version + nonce`, I think we don't need template type here.